### PR TITLE
fix: MA+YTMusic playback detection — wait for actual audio before starting round

### PR DESCRIPTION
**Pull request by Scribblerman**
## Summary
- Music Assistant with YouTube Music takes 5-15s to resolve URIs and buffer audio
- `blocking=True` on `music_assistant.play_media` hangs and times out after 8s
- Songs were skipped or rounds started before music was actually playing

### `services/media_player.py`
- MA: switched to `blocking=False` + state polling instead of `blocking=True`
- Poll waits for **actual playback** (not just queued): `media_title` changed + `media_position >= 1` + `media_position_updated_at` refreshed
- `position=0` means only "queued" in MA, `position >= 1` means speaker is outputting audio
- Returns `True` on timeout (don't skip song — MA may still be buffering)

### `game/state.py`
- Added `is_available()` check for **all platforms** (including MA) before `play_song()`
- Previously MA was excluded from pre-flight checks

### `tests/unit/test_media_player.py` (new)
- 10 tests covering: non-blocking call, song-change detection, position>=1 gate, 
  realistic YTMusic flow (queued→idle→playing pos=0→playing pos=1), timeout fallback,
  first song edge case, Sonos still uses blocking=True, availability checks

## Test plan
- [x] 287 unit tests passing (10 new)
- [x] Tested live with MA + Sonos + YouTube Music provider
- [x] Verified via HA state polling that `media_position >= 1` is the reliable indicator for actual audio output

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/unit/test_media_player.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    MediaPlayerService,
+)
+
+
+def _make_state(
+    state: str = "idle",
+    media_title: str = "Old Song",
+    media_position: float = 0,
+    media_position_updated_at: str = "2026-01-01T00:00:00+00:00",
+) -> MagicMock:
```
[the remark this answers is not in the document]
Comment (Scribblerman) [reply tests/unit/test_media_player.py]: Done

File: custom_components/beatify/services/media_player.py [hunk 2]
```diff
@@ -218,21 +218,64 @@ def _convert_uri_for_ma(uri: str) -> str:
...
+                    return True
+            await asyncio.sleep(0.5)
+            elapsed += 0.5
+
+        _LOGGER.warning(
+            "MA playback not confirmed after %.1fs for %s (state: %s)",
+            PLAYBACK_TIMEOUT,
+            uri,
+            self._hass.states.get(self._entity_id).state if self._hass.states.get(self._entity_id) else "unknown",
+        )
```
[the remark this answers is not in the document]
Comment (Scribblerman) [reply custom_components/beatify/services/media_player.py:278]: Done

File: custom_components/beatify/services/media_player.py [hunk 3]
```diff
@@ -218,21 +218,64 @@ def _convert_uri_for_ma(uri: str) -> str:
...
+        # Wait for the song to actually play on the speaker:
+        # - media_title changed (new song queued)
+        # - media_position is low (< 5s = song just started, not leftover from old song)
+        # - media_position_updated_at changed (speaker is actively reporting)
+        elapsed = 0.0
+        while elapsed < PLAYBACK_TIMEOUT:
+            state = self._hass.states.get(self._entity_id)
+            if state and state.state == "playing":
+                current_title = state.attributes.get("media_title", "")
+                position = state.attributes.get("media_position", 999)
```
[the remark this answers is not in the document]
Comment (Scribblerman) [reply custom_components/beatify/services/media_player.py]: Done

File: custom_components/beatify/game/state.py [hunk 4]
```diff
@@ -1491,9 +1491,19 @@ async def start_round(self, hass: HomeAssistant, _retry_count: int = 0) -> bool:
...
-            # Skip for MA players since they use music_assistant.play_media service
-            # which handles speaker state differently
+            # Pre-flight check: verify speaker is available before playing
+            if not self._media_player_service.is_available():
+                error_detail = f"Media player {self.media_player} is unavailable"
+                self.last_error_detail = error_detail
+                _LOGGER.error(
+                    "Media player not available: %s, pausing game",
+                    error_detail,
+                )
```
[the remark this answers is not in the document]
Comment (Scribblerman) [reply custom_components/beatify/game/state.py]: Done
